index.js

```diff
@@ -4,73 +4,85 @@
-    var re = /{([0-9a-zA-Z_$\._]+)}/g;
+    function dataDriven(data, fn) {
+        var mochaIt = it;
+        var mochaBefore = before;
+        // Regex used to find tokens, e.g. {foo.bar}, {foo}
+        var re = /{([0-9a-zA-Z_$\._]+)}/g;
 
-    data.forEach(function(testData) {
-        try {
-            it = function(title, f) {
-                var result;
+        data.forEach(function(testData) {
+            try {
+                it = function(title, f) {
+                    var result;
 
-                for (var key in testData) {
-                    if (testData.hasOwnProperty(key)) {
-                        while (result = re.exec(title)) {
-                            title = replaceTitleTokens(result, testData, title);
+                    for (var key in testData) {
+                        if (testData.hasOwnProperty(key)) {
+                            while (result = re.exec(title)) {
+                                title = replaceTitleTokens(result, testData, title);
+                            }
                         }
                     }
-                }
 
-                if (f !== undefined) {
+                    if (f !== undefined) {
+                        var testFn = f.length < 2 ?
+                            function() {
+                                return f.call(this,testData)
+                            } :
+                            function(done) {
+                                return f.call(this,testData,done)
+                            }
+                    }
+
+                    mochaIt(title, testFn);
+                };
+
+                before = function(f) {
                     var testFn = f.length < 2 ?
                         function() {
                             return f.call(this,testData)
                         } :
                         function(done) {
                             return f.call(this,testData,done)
                         }
-                }
-
-                mochaIt(title, testFn);
-            };
 
-            before = function(f) {
-                var testFn = f.length < 2 ?
-                    function() {
-                        return f.call(this,testData)
-                    } :
-                    function(done) {
-                        return f.call(this,testData,done)
-                    }
+                    mochaBefore(testFn);
+                };
 
-                mochaBefore(testFn);
-            };
+                fn();
+            } finally {
+                it = mochaIt;
+                before = mochaBefore;
+            }
+        })
+    }
 
-            fn();
-        } finally {
-            it = mochaIt;
-            before = mochaBefore;
-        }
-    })
-};
+    if (typeof module !== 'undefined' && module.exports) {
+        module.exports = dataDriven;
+    } else if (typeof define === 'function' && define.amd) {
+        define('data-driven', [], function () {
```

Comment:
Not sure whether it's good idea or not to specify mocha dependency here. Mocha doesn't seem to support requirejs (though I haven't searched deeply).